base_agent/dialogue_manager.py

```diff
@@ -38,30 +36,26 @@ class DialogueManager(object):
         agent: a droidlet agent
         model: a (perhaps ML) model used by and the model used for manager.
     """
-    def __init__(self, agent, model):
+
+    def __init__(
+        self, agent, dialogue_object_classes, opts, semantic_parsing_model_wrapper, model=None
+    ):
         self.agent = agent
         self.dialogue_stack = DialogueStack(agent, agent.memory)
         self.model = model
+        self.semantic_parsing_model_wrapper = semantic_parsing_model_wrapper(
+            agent=self.agent,
+            dialogue_object_classes=dialogue_object_classes,
+            opts=opts,
+            dialogue_manager=self,
+        )
 
-    def get_safety_words(self, safety_words_path):
-        """Read a list of safety words to prevent abuse.
-        """
-        with open(safety_words_path) as f:
-            safety_lines = f.readlines()
-        safety_words = []
-        for l in safety_lines:
-            w = l.strip("\n").lower()
-            if w != "" and w[0] != "<" and w[0] != "#":
-                safety_words.append(w)
-        return safety_words
+    def get_last_m_chats(self, m=1):
+        # fetch last m chats from memory
+        all_chats = self.agent.memory.get_recent_chats(n=m)
+        chat_list_text = [chat.chat_text for chat in all_chats]
 
-    def is_safe(self, chat):
-        """Check that chat does not contain unsafe words.
-        """
-        safety_set = set(self.safety_words)
-        cmd_set = set(chat.lower().split())
-        notsafe = len(cmd_set & safety_set) > 0
-        return not notsafe
+        return chat_list_text
```

Comment:
should return (chat_text, speaker) tuples, no?

Reply:
So we already get the speaker info with the chat: https://github.com/facebookresearch/droidlet/pull/314/files/4ff216cb3cafdb5053fd2caa5443ae1054d50d95#diff-79904ff197524a1a7d7aab8130ac6227e88fea0ddef7148377e9cb7503f8ef30R77 

and pass it along to get_dialogue_object() of the model wrapper.

This function only fetches the last m chats (plain text)

Reply:
but that is only the speaker for the incoming chat, no?  presumably if someone is using the multiturn chat history in their semantic parser, they will need the speaker for each of the chats in the history.

Reply:
Bu the speaker is going to be the same in the session(regardless of chat history)right ?

Reply:
Nevermind, I can extract the speaker from here. Let me push a commit